### PR TITLE
feat(env): Add the 'environment' property

### PR DESCRIPTION
**Pull request by jmauro**
# Description

I would like to add a new property 'env' to the resource

STATE:
There is currently no way to set some environment variables, see the
ENVIRONEMENT section in 'git-config'.

PROPOSED SOLUTION:
Add the 'env' property, and merge 'cmd_env' to use during the
Mixlib::ShellOut.

## Check List

- [X] A summary of changes made is included in the CHANGELOG under `## Unreleased`
- [/] New functionality includes testing.
- [/] New functionality has been documented in the README if applicable.

### CODE REVIEW

**Comment by jmauro** [comment]
Hello,

I don't think the error message is related to my PR:
```
Run actionshub/chef-delivery@main
  with:
  env:
    CHEF_LICENSE: accept-no-persist
/usr/bin/docker run --name b317a13[2](https://github.com/sous-chefs/git/runs/6272277663?check_suite_focus=true#step:4:2)d2275d84c8c90c71e2f88977e[3](https://github.com/sous-chefs/git/runs/6272277663?check_suite_focus=true#step:4:3)0_50859[4](https://github.com/sous-chefs/git/runs/6272277663?check_suite_focus=true#step:4:4) --label 294b31 --workdir /github/workspace --rm -e CHEF_LICENSE -e INPUT_GEMS -e HOME -e GITHUB_JOB -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_REPOSITORY_OWNER -e GITHUB_RUN_ID -e GITHUB_RUN_NUMBER -e GITHUB_RETENTION_DAYS -e GITHUB_RUN_ATTEMPT -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_SERVER_URL -e GITHUB_API_URL -e GITHUB_GRAPHQL_URL -e GITHUB_REF_NAME -e GITHUB_REF_PROTECTED -e GITHUB_REF_TYPE -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e GITHUB_ACTION_REPOSITORY -e GITHUB_ACTION_REF -e GITHUB_PATH -e GITHUB_ENV -e GITHUB_STEP_SUMMARY -e RUNNER_OS -e RUNNER_ARCH -e RUNNER_NAME -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e ACTIONS_CACHE_URL -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" -v "/home/runner/work/git/git":"/github/workspace" 294b31:7a132d227[5](https://github.com/sous-chefs/git/runs/6272277663?check_suite_focus=true#step:4:5)d84c8c90c[7](https://github.com/sous-chefs/git/runs/6272277663?check_suite_focus=true#step:4:7)1e2f88977e30

The Delivery CLI has been deprecated as part of the EOL of Chef Workflow (Delivery) on December 2020.
```

Could anyone help me on this one?

Kind regards,
JM

**Comment by damacus** [comment]
Hey @jmauro thanks for the PR. Let me copy to workflows from the Java cookbook. That should fix this error.

File: resources/config.rb [hunk 1]
```diff
@@ -15,7 +16,10 @@ def initialize(*args)
 end
 
 load_current_value do
-  cmd_env = user ? { 'USER' => user, 'HOME' => ::Dir.home(user) } : nil
+  cmd_env = user ? { 'USER' => user, 'HOME' => ::Dir.home(user) } : {}
+  if property_is_set?(:env)
+    cmd_env.merge!(env)
+  end
```
[the remark this answers is not in the document]
Comment (damacus) [reply resources/config.rb:22]: This is a more "Rubyist" way of doing the same thing 👍🏼

[the remark this answers is not in the document]
Comment (jmauro) [reply resources/config.rb:22]: Hello @damacus for the feedback.

My first intend when creating the PR was to be able to use the different git config [environment variable](https://git-scm.com/docs/git-config#_environment) to change the `.gitconfig` file. Unfortunately, it doesn't seem to work as expected when testing with kitchen-test. So I will close this PR and create a new one with the scope `file`, do you think this is legit?

Kind regards,
JM

**Pull request closed without merging**